superset/views/core.py

```diff
@@ -2093,8 +2093,20 @@ def sqllab_viz(self) -> FlaskResponse:  # pylint: disable=no-self-use
             .filter_by(database_id=database_id, table_name=table_name)
             .one_or_none()
         )
-        if not table:
-            table = SqlaTable(table_name=table_name, owners=[g.user])
+
+        if table:
+            return json_errors_response(
+                [
+                    SupersetError(
+                        message=f"Dataset [{table_name}] already exist",
```

Comment:
message=f"Dataset [{table_name}] already exist" <--- should this read 
message=f"Dataset [{table_name}] already exists" <--- exists instead of exist